arches/app/datatypes/datatypes.py

```diff
@@ -2114,6 +2124,11 @@ def to_json(self, tile, node):
     def collects_multiple_values(self):
         return True
 
+    def clean(self, tile, nodeid):
```

Comment:
Nice catch adding clean to the other list datatypes. The ResourceInstanceDataType is actually a list as well (just always limited to one entry). If this clean method was moved to ResourceInstanceDataType, then it would inherit in ResourceInstanceListDataType covering both.